lib/src/compile.ts

```diff
@@ -89,8 +91,14 @@ function newCompileStringRequest(
 
   if (options?.url) input.setUrl(options.url.toString());
 
-  if (options && 'importer' in options && options.importer) {
-    input.setImporter(importers.register(options.importer));
+  const hasImporter = options && 'importer' in options && options.importer;
+  if (options?.url && !hasImporter) {
+    const importer = new proto.InboundMessage.CompileRequest.Importer();
+    importer.setPath(p.resolve('.'));
+    input.setImporter(importer);
+  } else {
+    const importer = hasImporter ? options.importer : noOpImporter;
```

Comment:
Explicitly send a no-op importer so that imports will not be resolved by the otherwise default `FilesystemImporter('.')` on the compiler side when both `url` and `importer` are `null`.